packages/dds/tree/src/tree/anchorSet.ts

```diff
@@ -104,41 +104,124 @@ export class AnchorSet {
+     * Add an API to resurrect them? Store them in special detached fields? Store them in special non-detached fields?
+     *
+     * TODO:
+     * Now should custom anchors work (ex: ones not just tied to a specific Node)?
+     * This design assumes they can be expressed in terms of a Node anchor + some extra stuff,
+     * but we don't have an API for the extra stuff yet.
      */
     public moveChildren(
-        src: UpPath, srcField: FieldKey, start: number, count: number,
-        dst: UpPath, dstField: FieldKey, dstIndex: number): void {
-            const srcPath = this.find(src);
+        count: number,
+        src: undefined | { path: UpPath; field: FieldKey; start: number; },
+        dst: undefined | { path: UpPath; field: FieldKey; start: number; },
+    ): void {
+        assert(
+            src !== undefined || dst !== undefined,
+            "moveChildren is a no-op and should not be called if there is no src or dst",
+        );
+
+        const srcParent = src === undefined ? undefined : this.find(src.path);
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const srcChildren = srcParent?.children?.get(src!.field);
             // Sorted list of PathNodes to move from src to dst.
             let toMove: PathNode[];
-            if (srcPath !== undefined) {
-                // TODO: set toMove
-                // Remove items from src.
+
+        // Update src
+        if (srcChildren !== undefined) {
+            let numberBeforeMove = 0;
+            let numberToMove = 0;
+            for (let index = 0; index < srcChildren.length; index++) {
+                const element = srcChildren[index];
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                if (element.parentIndex < src!.start) {
+                    numberBeforeMove = index;
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                } else if (element.parentIndex < src!.start + count) {
+                    numberToMove++;
+                } else {
                 // Fix indexes in src after moved items (subtract count).
+                    srcChildren[index].parentIndex -= count;
+                }
+            }
+            toMove = srcChildren.splice(numberBeforeMove, numberToMove);
+            if (srcChildren.length === 0) {
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                srcParent!.afterEmptyField(src!.field);
+            }
                 toMove = [];
             } else {
                 toMove = [];
             }
+
+        if (dst === undefined) {
+            // Change is a delete.
+            // Moved items have already been un-parented, so just mark them as deleted.
+            for (const moved of toMove) {
+                assert(!moved.deleted, "PathNode must not be deleted");
+                moved.deleted = true;
```

Comment:
Consider exposing `PathNode.deleteThis()` and calling it instead.

Reply:
That does not work here: that demoves the node from its parent, which has already been done.

Note that we also can't just replace all this function in the delete case with deleteThis either, since we need the above logic to adjust indexes of siblings (deleteThis removes the tracking of an anchor, and does not handle any changes to the actual tree like this does in the delete case).